test/dynamo/test_sdpa.py

```diff
@@ -0,0 +1,106 @@
+# Owner(s): ["module: dynamo"]
+import contextlib
+
+import torch._dynamo.test_case
+import torch._dynamo.testing
+from torch._dynamo.testing import CompileCounter
+from torch.backends.cuda import SDPAParams
+
+
+@contextlib.contextmanager
+def allow_in_graph_sdpa_params():
+    global SDPAParams
+    try:
+        old = SDPAParams
+        SDPAParams = torch._dynamo.allow_in_graph(SDPAParams)
+        yield
+    finally:
+        SDPAParams = old
+
+
+class TestSDPA(torch._dynamo.test_case.TestCase):
+    def assert_ref_equals_params(self, actual, expected):
+        self.assertIs(actual.query, expected.query)
+        self.assertIs(actual.key, expected.key)
+        self.assertIs(actual.value, expected.value)
+        self.assertIs(actual.attn_mask, expected.attn_mask)
```

Comment:
This just sounds like comparison isnt implemented properly for the object? Why cant we do `self.assertIs/self.assertEqual(actual, expected)`? why do we need to decompose it like this?

Reply:
The instance of SDPAParams will be different possibly since we construct new instances on output (ie in the event of a graph break we reconstruct), but the refs inside it should be the same, this isn't covered by either of your suggestions.